woocommerce/woocommerce-ios
# Add-ons: Enable feature through the beta features screen

# Why

The order add-ons workaround is now complete and usable. This PR makes sure that the feature is accessible by enabling it through the beta-feature screen that lives in the settings menu.

# How

- Update `GeneraleAppSettings` with a new property `isViewAddOnsSwitchEnabled` to store the most recent user preference.

- Generate `copiable` methods for `GeneralAppSettings`

- Created two new actions in `AppSettingsAction` to be able to set and load the user preference for the "order add-on" beta feature switch.

- Update `OrderDetailsDataSource` to expose the `showAddOns` property so it can be modified depending on the user "order add-on switch" preference.

- Update `OrderDetailsViewModel` to fetch the latest user preference and set `OrderDetailsDataSource.showAddOns` accordingly.

- Update `BetaFeatureViewController` to change the "create product" row for an "viewing add-on" row.

- Update `SettingsViewController` to enable showing the beta features screen.


# Demo

https://user-images.githubusercontent.com/562080/117145871-3bbfb680-ad79-11eb-8912-b291195dd3f2.mov

# Testing steps
**Prerequisites:** Have your site with the add-ons plugin installed and with at least one order with add-ons.

- Launch the app and navigate to the add-on order
- See that the order does not display any "View add-on" button
- Go back to settings -> experimental features and toggle the order add-ons switch
- Go back to the order
- See that the  "View Add-Ons" button is present.


Update release notes:
